### PR TITLE
[12.0][ADD] stock_location_empty

### PR DESCRIPTION
**Pull request by TDu**

### CODE REVIEW

File: stock_location_empty/models/stock_location.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    def _search_location_amount(self, operator, value):
+        if operator not in ('=', '!=', '<', '<=', '>', '>='):
+            return []
+        # pylint: disable=sql-injection
+        self.env.cr.execute(
+            """
+            SELECT loc.id FROM stock_location loc
+            LEFT OUTER JOIN stock_quant quant ON loc.id = quant.location_id
+            GROUP BY loc.id
+            HAVING coalesce(sum(quantity), 0) %s %%s;"""
```
Comment (i-vyshnevska) [inline stock_location_empty/models/stock_location.py]:
```suggestion
            HAVING coalesce(sum(quantity), 0) %s %%s;"""
```
is it correct?

**Review by i-vyshnevska — APPROVED** [review]

File: stock_location_empty/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+{'name': 'Stock Location Empty',
```
Comment (rousseldenis) [inline stock_location_empty/__manifest__.py:4]: Can you update this according to https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py ?

File: stock_location_empty/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ 'version': '12.0.1.0.0',
+ 'author': 'Camptocamp, Odoo Community Association (OCA)',
+ 'license': 'AGPL-3',
+ 'category': 'stock',
```
Comment (rousseldenis) [inline stock_location_empty/__manifest__.py:8]: Change this to 'Warehouse' as it won't appear in the good category in OCA shop

File: stock_location_empty/models/stock_location.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models
+
+
+class StockLocation(models.Model):
+    _inherit = 'stock.location'
+
+    stock_amount = fields.Integer(
```
Comment (rousseldenis) [inline stock_location_empty/models/stock_location.py:11]: I'd prefer stock_quantity as amount is more related to monetary.
Moreover, this field should be Float

File: stock_location_empty/models/stock_location.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        compute='_compute_location_amount', search='_search_location_amount'
+    )
+
+    @api.multi
+    def _search_location_amount(self, operator, value):
+        if operator not in ('=', '!=', '<', '<=', '>', '>='):
+            return []
+        # pylint: disable=sql-injection
```
Comment (rousseldenis) [inline stock_location_empty/models/stock_location.py:19]: Don't do that. Pylint error says that you have to pass parameters to second argument:

`cr.execute("blabla %s", value)`

File: stock_location_empty/models/stock_location.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        self.env.cr.execute(
```
Comment (rousseldenis) [inline stock_location_empty/models/stock_location.py:20]: Why not using read_group ???

File: stock_location_empty/models/stock_location.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            HAVING coalesce(sum(quantity), 0) %s %%s;"""
+            % operator,
+            (value,),
+        )
+        ids = [row[0] for row in self.env.cr.fetchall()]
+        return [('id', 'in', ids)]
+
+    @api.multi
+    def _compute_location_amount(self):
+        self.env.cr.execute(
```
Comment (rousseldenis) [inline stock_location_empty/models/stock_location.py:34]: Same as above, use read_group

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

**Comment by yvaucher** [comment]
@TDu ping

**Pull request closed without merging**